crates/libcontainer/src/process/container_intermediate_process.rs

```diff
@@ -115,7 +118,13 @@ pub fn container_intermediate_process(
         .close()
         .context("failed to close unused init sender")?;
 
-    Ok(())
+    if wait {
```

Comment:
Can we just return the pid here and then do the wait in container_main_process (line 36)? I would like to handle this as high up as possible and leave the lower layers untouched.

Reply:
Hey, @Furisto that was the original way I tried to do it, but the issue with it is that the final process (container process) is child of the intermediate process, so only that can wait for it. Trying to `waitpid` on the final process from the main process fails, as it is not main process's child. Thus we need to make intermediate wait for the final process, and main process wait for intermediate

Reply:
I understand that. What I am asking is to do the wait [here](https://github.com/containers/youki/blob/0f3d89f5e70f68c8491aa84e9d2278fb15c7f306/crates/libcontainer/src/process/container_main_process.rs#L36), that's in the main_process file, but the code in that location is executed by the intermediate process.

Reply:
Oh, ok, sorry I misunderstood earlier. Can you take a look at https://github.com/utam0k/youki/pull/116 and see if that is what you expected?